File: 프로그래머스/Level1/체육복.py
# https://programmers.co.kr/learn/courses/30/lessons/42862
def solution(n, lost, reserve):
    count =0
    set_lost = set(lost)-set(reserve)
    set_reserve = set(reserve)-set(lost)
    for i in set_lost:
        if i-1 in set_reserve:
            set_reserve.remove(i-1)
            count+=1
        elif i+1 in set_reserve:
            set_reserve.remove(i+1)
            count+=1
    answer = n-len(set_lost)+count
    return answer

File: 프로그래머스/Level1/체육복.js
function solution(n, lost, reserve) {
  let filtered_lost = lost.filter( x => !reserve.includes(x));
  let filtered_reserve = reserve.filter( x => !lost.includes(x));
  filtered_lost = filtered_lost.sort((a,b)=>{return a-b})
  filtered_reserve = filtered_reserve.sort((a,b)=>{return a-b})
  let answer = n-filtered_lost.length;
  filtered_lost.forEach( num =>{
      if(filtered_reserve.includes(num-1)){
          filtered_reserve = filtered_reserve.filter( element => element !== (num-1));
          answer+=1
      }else if(filtered_reserve.includes(num+1)){
          filtered_reserve = filtered_reserve.filter( element => element !== (num+1));
          answer+=1
      }else{
          answer+=0
      }
  })
  return answer;
}

//  차집합, 정렬, includes, forEach